### PR TITLE
dd.4.18 adminless install for mac

### PR DESCRIPTION
**Pull request by aevesdocker**

### CODE REVIEW

File: desktop/install/mac-install.md [hunk 1]
```diff
@@ -78,11 +78,15 @@ Your Mac must meet the following requirements to install Docker Desktop successf
...
-4. Select **Accept** to continue. Docker Desktop starts after you accept the terms.
+4. Select **Accept** to continue. 
 
    Note that Docker Desktop will not run if you do not agree to the terms. You can choose to accept the terms at a later date by opening Docker Desktop.
 
    For more information, see [Docker Desktop Subscription Service Agreement](https://www.docker.com/legal/docker-subscription-service-agreement){: target="_blank" rel="noopener" class="_" }. We recommend that you also read the [FAQs](https://www.docker.com/pricing/faq){: target="_blank" rel="noopener" class="_"}.
+5. From the installation window, select either: 
+   - **Use recommended settings (Requires password)**. This let's Docker Desktop automatically set the necessary configuration settings. 
+   - **Use advanced settings**. You can then set the location of the Docker CLI tools either in the system or user directory, enable the default Docker socket, and enable privileged port mapping. See [Settings](../settings/mac.md#advanced), for more information.
+6. Select **Finish**. If you have applied any of the above configurations in step 5, you are asked to enter your password to confirm.
```
Comment (aiordache) [inline desktop/install/mac-install.md:89]: The password prompt appears only if the options requiring password are selected.

File: desktop/settings/mac.md [hunk 2]
```diff
@@ -50,6 +50,25 @@ On the **General** tab, you can configure when to start Docker and specify other
...
+  - **System**: Docker CLI tools are installed in the system directory under `/usr/local/bin`
+  - **User**: Docker CLI tools are installed in the user directory under `$HOME/.docker/bin`. You must then add `$HOME/.docker/bin` to your PATH. To add `$HOME/.docker/bin` to your path, run:
+
+   ```console
+   $ export PATH=$PATH:~/.docker/bin
+   ```
+
+- **Enable default Docker socket (Requires password)**. Creates `/var/run/docker.sock` which some third party clients may use to communicate with Docker Desktop. For more information, see [permission requirements for macOS](../mac/permission-requirements.md#installing-symlinks)
+
+- **Enable privileged port mapping (Requires password)**. Starts the privileged ports that are between 1 and 1024. For more information, see [permission requirements for macOS](../mac/permission-requirements.md#binding-privileged-ports)
```
Comment (aiordache) [inline desktop/settings/mac.md:67]: Starts the privileged helper process which binds the ports ...

File: desktop/mac/permission-requirements.md [hunk 3]
```diff
@@ -14,7 +14,7 @@ It also provides clarity on running containers as `root` as opposed to having `r
 ## Permission requirements
 
 Docker Desktop for Mac is run as an unprivileged user. However, certain functionalities may be required for Docker Desktop to perform a limited set of privileged configurations such as:
- - [Installing symlinks](#installing-symlinks) in `/usr/local/bin`. This ensures the `docker` CLI is on the user’s PATH without having to reconfigure shells, log out then log back in, for example.
+ - [Installing symlinks](#installing-symlinks) in either `/usr/local/bin` or `$HOME/.docker/bin`.
```
Comment (MihaelaStoica) [inline desktop/mac/permission-requirements.md:17]: installing in  `$HOME/.docker/bin` is not a privileged operation

File: desktop/mac/permission-requirements.md [hunk 4]
```diff
@@ -60,6 +125,10 @@ ERRO[0003] error waiting for container: context canceled
 >
 > The command may fail with the same error if the user takes too long to authorize the prompt to start the helper process, as it may timeout.
 
+<hr>
+</div>
+</div>
+
 ### Ensuring `localhost` and `kubernetes.docker.internal` are defined
```
Comment (aevesdocker) [inline desktop/mac/permission-requirements.md:132]: @MihaelaStoica does this need to change in any way for 4.18?

File: desktop/mac/permission-requirements.md
Comment (aevesdocker) [inline desktop/mac/permission-requirements.md:1]: I think the layout of this page is better now @MihaelaStoica. Preview here:

https://deploy-preview-16852--docsdocker.netlify.app/desktop/mac/permission-requirements/

File: desktop/mac/permission-requirements.md (hunk 4 above)
Comment (MihaelaStoica) [reply desktop/mac/permission-requirements.md:132]: Yes, we are not doing this anymore, so perhaps we can say that it's now the user's responsibility to ensure that
`localhost` is resolved to `127.0.0.1` and , if kubernetes is used, that `kubernetes.docker.internal` is resolved to `127.0.0.1`

@aiordache please confirm

**Review by MihaelaStoica — CHANGES_REQUESTED** [review]

File: desktop/mac/permission-requirements.md [hunk 5]
```diff
@@ -14,34 +14,99 @@ It also provides clarity on running containers as `root` as opposed to having `r
...
 
-Installing symlinks in `/usr/local/bin` is a privileged configuration Docker Desktop performs on the first startup. Docker Desktop checks if symlinks exists and takes the following actions: 
+With version 4.18 and later users can choose whether symlinks are installed either in `/usr/local/bin` or `$HOME/.docker/bin` during installation of Docker Desktop.
+
+If `/usr/local/bin` is chosen, Docker Desktop requires authorization to confirm this choice before the symlinks to docker binaries are created in `/usr/local/bin`. If `$HOME/.docker/bin` is chosen, authorization is not required, but the user must [manually add `$HOME/.docker/bin`](../settings/mac.md#advanced)to their PATH. 
+
+Users are also given the option to enable the installation of the `/var/run/docker.sock` symlink. Creating this symlink ensures various docker clients relying on the default docker socket path to work without additional changes. As the `/var/run` is mounted as a tmpfs, its content is deleted on restart, symlink to docker socket included.
```
[the remark this answers is not in the document]
Comment (MihaelaStoica) [reply desktop/mac/permission-requirements.md:72]: see this together with the suggestion on line 74 below

**Comment by aevesdocker** [comment]
Thank you for the comments @MihaelaStoica. Everything has been updated

File: desktop/mac/permission-requirements.md [hunk 6]
```diff
@@ -14,34 +14,99 @@ It also provides clarity on running containers as `root` as opposed to having `r
...
-The only time Docker Desktop may start the privileged process is for binding privileged ports that are not allowed by default on the host OS.
+<hr>
+</div>
+</div>
 
 ### Installing symlinks
 
-The docker binaries are installed by default in `/Applications/Docker.app/Contents/Resources/bin`. Docker Desktop ensures the `docker` CLI is on the user’s PATH without having to reconfigure shells, log out then log back in for example. As on most systems `/usr/local/bin` is in the user's PATH by default, and so Docker Desktop creates symlinks for all docker binaries in it.
+The docker binaries are installed by default in `/Applications/Docker.app/Contents/Resources/bin`. Docker Desktop ensures the docker CLI is on the user’s PATH without having to reconfigure shells, log out then log back in for example. As on most systems `/usr/local/bin` is in the user's PATH by default, and so Docker Desktop creates symlinks for all docker binaries in it.
```
Comment (dvdksn) [inline desktop/mac/permission-requirements.md:61]:
```suggestion
The Docker binaries are installed by default in `/Applications/Docker.app/Contents/Resources/bin`. Docker Desktop ensures the Docker CLI is on the user’s PATH without having to reconfigure shells, log out then log back in for example. As on most systems `/usr/local/bin` is in the user's PATH by default, and so Docker Desktop creates symlinks for all Docker binaries in it.
```

File: desktop/mac/permission-requirements.md [hunk 7]
```diff
@@ -14,34 +14,99 @@ It also provides clarity on running containers as `root` as opposed to having `r
...
+  <li class="active"><a data-toggle="tab" data-target="#tab4">Version 4.18 and later</a></li>
+  <li><a data-toggle="tab" data-target="#tab5">Version 4.17 and earlier </a></li>
+</ul>
+<div class="tab-content">
+<div id="tab4" class="tab-pane fade in active" markdown="1">
+
+With version 4.18 and later users can choose whether symlinks are installed either in `/usr/local/bin` or `$HOME/.docker/bin` during installation of Docker Desktop.
 
-Installing symlinks in `/usr/local/bin` is a privileged configuration Docker Desktop performs on the first startup. Docker Desktop checks if symlinks exists and takes the following actions: 
+If `/usr/local/bin` is chosen and this location is not writable by unprivileged users, Docker Desktop requires authorization to confirm this choice before the symlinks to docker binaries are created in `/usr/local/bin`. If `$HOME/.docker/bin` is chosen, authorization is not required, but the user must [manually add `$HOME/.docker/bin`](../settings/mac.md#advanced)to their PATH.
```
Comment (dvdksn) [inline desktop/mac/permission-requirements.md:72]:
```suggestion
If `/usr/local/bin` is chosen, and this location is not writable by unprivileged users, Docker Desktop requires authorization to confirm this choice before the symlinks to Docker binaries are created in `/usr/local/bin`. If `$HOME/.docker/bin` is chosen, authorization is not required, but the user must [manually add `$HOME/.docker/bin`](../settings/mac.md#advanced) to their PATH. 
```

File: desktop/mac/permission-requirements.md [hunk 8]
```diff
@@ -14,34 +14,99 @@ It also provides clarity on running containers as `root` as opposed to having `r
...
 - Creates the symlinks without the admin prompt if `/usr/local/bin` is writable by unprivileged users.
-- Triggers an admin prompt for the user to authorize the creation of symlinks in `/usr/local/bin`. If the user authorizes this, symlinks to docker binaries are created in `/usr/local/bin`.
-If the user rejects the prompt, is not willing to run configurations requiring elevated privileges, or does not have admin rights on their machine, Docker Desktop creates the symlinks in `~/.docker/bin` and edits the user's shell profile to ensure this location is in the user's PATH. This requires all open shells to be reloaded. 
+- Triggers an admin prompt for the user to authorize the creation of symlinks in `/usr/local/bin`. If the user authorizes this, symlinks to docker binaries are created in `/usr/local/bin`. If the user rejects the prompt, is not willing to run configurations requiring elevated privileges, or does not have admin rights on their machine, Docker Desktop creates the symlinks in `~/.docker/bin` and edits the user's shell profile to ensure this location is in the user's PATH. This requires all open shells to be reloaded.
```
Comment (dvdksn) [inline desktop/mac/permission-requirements.md:84]:
```suggestion
- Triggers an admin prompt for the user to authorize the creation of symlinks in `/usr/local/bin`. If the user authorizes this, symlinks to Docker binaries are created in `/usr/local/bin`. If the user rejects the prompt, is not willing to run configurations requiring elevated privileges, or does not have admin rights on their machine, Docker Desktop creates the symlinks in `~/.docker/bin` and edits the user's shell profile to ensure this location is in the user's PATH. This requires all open shells to be reloaded. 
```

**Review by dvdksn — APPROVED** [review]

File: desktop/mac/permission-requirements.md [hunk 9]
```diff
@@ -14,34 +14,99 @@ It also provides clarity on running containers as `root` as opposed to having `r
...
+
+<ul class="nav nav-tabs">
+  <li class="active"><a data-toggle="tab" data-target="#tab1">Version 4.18 and later</a></li>
+  <li><a data-toggle="tab" data-target="#tab2">Version 4.15 - 4.17 </a></li>
+  <li><a data-toggle="tab" data-target="#tab3">Versions prior to 4.15</a></li>
+</ul>
+<div class="tab-content">
+<div id="tab1" class="tab-pane fade in active" markdown="1">
+
+From version 4.18 and above of Docker Desktop, privileged access for certain functionalities is granted during installation. The first time Docker Desktop for Mac is launched, the user is presented with an installation window to either use the recommended settings, or configure:
```
Comment (christophermclellan) [inline desktop/mac/permission-requirements.md:32]: Hi @aevesdocker, Hi @MihaelaStoica, sorry I'm late with this feedback. I'd perhaps adjust to something like:

_"From version v4.18 and above, Docker Desktop for Mac provides more granular control over functionality that is enabled during installation. The first time Docker Desktop for Mac is launched, you’ll see an installation window prompting you to use the default settings, which sets configurations that work for most developers and requires that privileged access is granted.

If you work in an environment with elevated security requirements, such as where local administrative access is prohibited, then you can configure the Advanced Settings to remove the need for granting privileged access. The Advanced Settings allow you to configure:

* The location of the Docker CLI tools either in the system or user directory
* The default Docker socket
* Privileged port mapping.

These configurations can be changed at a later time from the Advanced page within Docker Desktop’s Settings."_

File: desktop/mac/permission-requirements.md [hunk 10]
```diff
@@ -14,34 +14,99 @@ It also provides clarity on running containers as `root` as opposed to having `r
...
+- The location of the Docker CLI tools either in the system or user directory
+- The default Docker socket
+- Privileged port mapping. 
+
+The user is then asked for their password to confirm.
```
Comment (christophermclellan) [inline desktop/mac/permission-requirements.md:37]: The user is not asked for their password if they change some of the Advanced settings. That's the point of this new flow, so we should adjust this.

**Review by dvdksn — APPROVED** [review]

**Pull request merged**